python_modules/libraries/dagster-duckdb/dagster_duckdb/io_manager.py

```diff
@@ -143,13 +149,20 @@ def _get_cleanup_statement(table_slice: TableSlice) -> str:
     Returns a SQL statement that deletes data in the given table to make way for the output data
     being written.
     """
-    if table_slice.partition:
-        partition_where = (
-            _static_where_clause(table_slice.partition)
-            if isinstance(table_slice.partition.partition, str)
-            else _time_window_where_clause(table_slice.partition)
-        )
-        return f"DELETE FROM {table_slice.schema}.{table_slice.table}\n" + partition_where
+    if table_slice.partition and len(table_slice.partition) > 0:
+        query = f"DELETE FROM {table_slice.schema}.{table_slice.table} WHERE\n"
+        for i in range(len(table_slice.partition)):
```

Comment:
building up the query as a bunch of string concatenations feels a bit fragile to me - are there better options? is that something sqlalchemy could be used for?